### PR TITLE
Add docs for ClusterGroupUpgrade states transitions

### PR DESCRIPTION
**Pull request by rcarrillocruz**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,7 +15,33 @@ The spec allows you to define:
...
+In order to model the various phases of an upgrade, the ClusterGroupUpgrade CR can be in the
+following states:
+
+* **UpgradeNotStarted**
+  * In this state, the **ClusterGroupUpgrade** CR has just been created and the *remediationAction* field is set to *inform*
+  * The controller will build a remediation plan based on the *clusters* list and with *remediationStrategy* fields. For example, if the *clusters* list is made of 6 clusters and the *remediationStrategy* has *maxConcurrency* set to 2 the controller will create a remediation plan with 3 batches: the first will contain the first two clusters, the second the third and the fourth and the third batch the fifth and the sixth. In another example, if the *clusters* list is made of the same 6 clusters and the *remediationStrategy* has *maxConcurrency* set to 2 but *canaries* list contains the second cluster from the *clusters* list then the controller will build a remediation plan of 4 batches: the first will contain the second cluster, the second batch will contain the first and the third, the third batch will contain the fourth and the fifth and the fourth batch will contain the sixth cluster.
```
Comment (irinamihai) [inline README.md:29]: "the second the third and the fourth and the third batch the fifth and the sixth" - This phrasing is a bit confusing. Maybe we could say "the second batch with contain clusters 3 & 4 and the 3rd batch clusters 5 & 6" or something similar.

File: README.md [hunk 2]
```diff
@@ -15,7 +15,33 @@ The spec allows you to define:
...
+  * The controller will transition to **UpgradeNotCompleted** once the *remediationAction* field is set to *enforce*
+* **UpgradeNotCompleted**
+  * In this state, the controller will enforce the policies following the remediation plan built in the **UpgradeNotStarted** state
+  * The policies of subsequent batches will start as soon as the current batch policies are all compliant unless the batch times out, in which case the controller will move on to the next batch. The value of this batch timeout is the **ClusterGroupUpgrade** timeout divided by the remediation plan number of batches.
+  * Within a batch, the platform upgrade will be enforced first. Once the platform upgrade policy is compliant, then the operator upgrades policies will be enforced
+  * The controller will transition to **UpgradeTimedOut** state in two cases:
+    * If the **ClusterGroupUpgrade** has the first batch as canaries and the policies for this first batch are not compliant within the batch timeout
+    * If the upgrade has exceeded the *timeout* value specified in the *remediationStrategy*
+* **UpgradeTimedOut**
```
Comment (irinamihai) [inline README.md:39]: What if some of the policies never become compliant?

Comment (rcarrillocruz) [reply README.md:39]: That is what the second bullet says, will rephrase

**Review by yrobla — APPROVED** [review]

**Pull request merged**